packages/core/helper-plugin/lib/src/components/RelationInput/components/Option.js

```diff
@@ -36,35 +36,26 @@ export const Option = (props) => {
       id: 'content-manager.components.Select.publish-info-title',
       defaultMessage: 'State: Published',
     };
+    const { isDraft } = props.data;
     const title = isDraft ? formatMessage(draftMessage) : formatMessage(publishedMessage);
 
     return (
       <Component {...props}>
         <Flex>
           <StyledBullet title={title} isDraft={isDraft} />
-          <Typography ellipsis>{props.label || '-'}</Typography>
+          <Typography ellipsis>{props.data.mainField || '-'}</Typography>
```

Comment:
we don't automatically get the `props.label` with this new data structure, not sure how it worked before 🤔 wdyt @gu-stav

Reply:
Before the normalization didn't happen. You destructure `{ mainField }` from `props.data` above, where you get `isDraft` and then re-use it here without `props.data`.

Could we please use:

```suggestion
          <Typography ellipsis>{props.data.mainField ?? '-'}</Typography>
```

instead, bcause `mainField` may contain `0`, which would render as `''` otherwise.